chain/actors/builtin/market/market.go

```diff
@@ -127,7 +127,7 @@ type State interface {
 	) (weight, verifiedWeight abi.DealWeight, err error)
 	NextID() (abi.DealID, error)
 	GetState() interface{}
-	GetAllocationIdForPendingDeal(dealId abi.DealID) (verifregtypes.AllocationId, error)
+	GetAllocationIdForPendingDeal(dealId abi.DealID) (*verifregtypes.AllocationId, error)
```

Comment:
Instead of using nil you shoulld use the zero value which stands for invalid allocation.  Updated go-state-types to help out: https://github.com/filecoin-project/go-state-types/pull/93